code/modules/vehicles/mecha/combat/savannah_ivanov.dm

```diff
@@ -0,0 +1,277 @@
+	layer = initial(layer)
+	skyfall_charge_level = 0
+	update_icon_state()
+	for(var/turf/open/floor/tiled_turf in range(1, src))
+		tiled_turf.break_tile()
+	var/turf/landed_on = get_turf(src)
+	if(isclosedturf(landed_on))
+		landed_on.ScrapeAway()
+	for(var/mob/living/victim in landed_on)
+		to_chat(victim, "<span class='userdanger'>[src] lands on you from above!</span>")
+		if(victim.stat != CONSCIOUS)
+			victim.gib(FALSE, FALSE, FALSE)
+		else
+			victim.adjustBruteLoss(80)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/abort_skyfall(mob/pilot)
+	if(pilot)
+		to_chat(pilot, "[icon2html(src, pilot)]<span class='notice'>Skyfall aborted.</span>")
+	COOLDOWN_START(src, skyfall_cooldown, skyfall_charge_level * 10 SECONDS) //so aborting skyfall later in the process imposes a longer cooldown
+	skyfall_charge_level = 0
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/start_missile_targeting(mob/gunner, silent = TRUE)
+	if(!silent)
+		to_chat(gunner, "<span class='warning'>Ivanov Strike targeting process booted.</br>\
+		Your next click will fire the missile (provided the mech is facing the right direction).</span>")
+	aiming_ivanov = TRUE
+	RegisterSignal(src, COMSIG_MECHA_MELEE_CLICK, .proc/on_melee_click)
+	RegisterSignal(src, COMSIG_MECHA_EQUIPMENT_CLICK, .proc/on_equipment_click)
+	gunner.client.mouse_override_icon = 'icons/effects/mouse_pointers/supplypod_down_target.dmi'
+	gunner.update_mouse_pointer()
+	gunner.overlay_fullscreen("ivanov", /atom/movable/screen/fullscreen/ivanov_display, 1)
+	SEND_SOUND(gunner, 'sound/machines/terminal_on.ogg') //spammable so I don't want to make it audible to anyone else
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/end_missile_targeting(mob/gunner, silent = TRUE)
+	if(!silent)
+		to_chat(gunner, "<span class='warning'>Ivanov Strike targeting process killed.</span>")
+	aiming_ivanov = FALSE
+	UnregisterSignal(src, list(COMSIG_MECHA_MELEE_CLICK, COMSIG_MECHA_EQUIPMENT_CLICK))
+	gunner.client.mouse_override_icon = null
+	gunner.update_mouse_pointer()
+	gunner.clear_fullscreen("ivanov")
+
+///signal called from clicking with no equipment
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/on_melee_click(datum/source, mob/living/pilot, atom/target, on_cooldown, is_adjacent)
+	SIGNAL_HANDLER
+	if(!target)
+		return
+	drop_missile(pilot, get_turf(target))
+
+///signal called from clicking with equipment
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/on_equipment_click(datum/source, mob/living/pilot, atom/target)
+	SIGNAL_HANDLER
+	if(!target)
+		return
+	drop_missile(pilot, get_turf(target))
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/drop_missile(mob/gunner, turf/target_turf)
+	end_missile_targeting(gunner)
+	SEND_SOUND(gunner, 'sound/machines/triple_beep.ogg')
+	COOLDOWN_START(src, strike_cooldown, strike_cooldown_time)
+	podspawn(list(
+		"target" = target_turf,
+		"style" = STYLE_MISSILE,
+		"effectMissile" = TRUE,
+		"explosionSize" = list(0,0,3,3)
+	))
```

Comment:
Just a suggestion, maybe you can make the missiles weaker but allow to drop 2-3 of them?

Reply:
Gonna sound like a baby here but if you want it, you should pr it to my branch